research/picai/README.md

```diff
@@ -94,35 +94,82 @@ The raw annotations differ in spatial resolution across annotators. Specifically
+```
+python research/picai/prepare_data.py --scans_read_dir /path --annotations_read_dir /path --scans_write_dir /path --overview_write_dir /path --size 20 256 256  --spacing 3 0.5 0.5 --num_threads 4 --splits_path /path
+```
+The arguments are defined accordingly:
+- **scans_read_dir**: Base directory containing a number of subdirectories - each with one or more scans.
+- **annotations_read_dir**: Base directory containing a number of annotations files.
+- **scans_write_dir**: Base directory to write all preprocessed scans.
+- **annotations_write_dir**: Base directory to write all preprocessed annotations.
+- **overview_write_dir**: Base directory to write all dataset overviews (more details below).
+- **size**: Desired dimensions of preprocessed images.
+- **spacing**: Desired spacing of preprocessed images.
+- **num_threads**: The number of threads to use during data preprocessing.
+- **splits_path**: Optional path to the desired splits. Defaults to official picai splits.
+
+The specific artifacts of the `prepare_data.py` script are the preprocessed scans and annotations saved to the specified write directories. Dataset overview(s) are also generated. These are based on the splits json file that is passed to the script. The dataset overview files include a list of sets of scan paths along with corresponding annotations for a specific fold.
+
+The default settings of the `prepare_data.py` script perform the following operations:
 - **Resampling Spatial Resolution:** The spatial resolution of its images vary across different patient exams. For the axial T2W scans, the most common voxel spacing (in mm/voxel) observed is 3.0×0.5×0.5 (43%), followed by 3.6×0.3×0.3 (25%), 3.0×0.342×0.342 (15%) and others (17%). As a naive approach, we simply resample all scans to 3.0×0.5×0.5 mm/voxel.
 
 - **Cropping to Region-of-Interest:** We naively assume that the prostate gland is typically located within the centre of every prostate MRI scan. Hence, we take a centre crop of each scan, measuring 20×256×256 voxels in dimensions.
 
 Each preprocessed image sample is stored on the cluster at the following path:
 ```
-/ssd003/projects/aieng/public/PICAI/workdir/nnUNet_raw_data/Task2203_picai_baseline/imagesTr/<patient_id>_<study_id>.nii.gz
+/ssd003/projects/aieng/public/PICAI/nnUNet/nnUNet_raw/Dataset003_PICAI/imagesTr/<patient_id>_<exam_id>_<modality_id>.nii.gz
 ```
 
-and the corresponding label
+Thus, all preprocessed images are written to the same directory. Each scan is uniquely identified with a **<patient_id>** and **<study_id>** described above, along with a **<modality_id>**. The **<modality_id>** specifies what type the scan is. In our case, the **<modality_id>** has the following mapping:
```

Comment:
Based on the example above, it looks like it's `exam_id` now rather than `study_id`?

Reply:
They are interchangeable but I did not realize I deviated from study_id so I will change all references to that